RenderTheWorld/RenderTheWorld.js

```diff
@@ -0,0 +1,1806 @@
+// 依赖库
+import * as THREE from './three.js';
+
+// 插件
+// 这里为该扩展目录下的三个js文件
+import { OBJLoader } from './OBJLoader.js';
+import { OrbitControls } from './OrbitControls.js';
+import { MTLLoader } from './MTLLoader.js';
+import { GLTFLoader } from './GLTFLoader.js';
+// import {OBJLoader} from 'https://cdn.jsdelivr.net/gh/MoreBugOfDog/cdn-file@TBS/file/OBJLoader.js';
+// import {OrbitControls} from 'https://cdn.jsdelivr.net/gh/MoreBugOfDog/cdn-file@TBS/file/OrbitControls.js';
+// import {MTLLoader} from 'https://cdn.jsdelivr.net/gh/MoreBugOfDog/cdn-file@TBS/file/MTLLoader.js';
+import WebGL from './WebGL.js';
+import {
+	chen_RenderTheWorld_picture,
+	chen_RenderTheWorld_icon,
+} from './assets/index.js';
+
+// console.log(THREE);
+
+const chen_RenderTheWorld_extensionId = 'RenderTheWorld';
+
+/** @typedef {string|number|boolean} SCarg 来自Scratch圆形框的参数，虽然这个框可能只能输入数字，但是可以放入变量，因此有可能获得数字、布尔和文本（极端情况下还有 null 或 undefined，需要同时处理 */
+
+class RenderTheWorld {
+	constructor(runtime) {
+		this.runtime = runtime;
+		// console.log(this.runtime);
+		this.__draw = this.runtime.renderer.draw;
+		this.__resize = this.runtime.renderer.resize;
```

Comment:
```suggestion
                const _draw = this.runtime.renderer.draw;
                const _resize = this.runtime.renderer.resize;
		this.runtime.renderer.resize = (pixelsWide, pixelsTall) => {
			_resize.call(this.runtime.renderer, pixelsWide, pixelsTall);
			this.tc.style.width = pixelsWide;
		        this.tc.style.height = pixelsTall;
		};
		this.runtime.renderer.draw = () => {
			if (!this.isTcShow) {
				_draw.call(this.runtime.renderer);
			} else if (this.dirty) {
			        this.dirty = false; // TODO: 和 Scratch renderer 共用 dirty
			        // this.dirty 是一个变量，每当场景变更（需要渲染）时就设置为 true
			        this.renderer.render(this.scene, this.camera);
			}
		};
		this.dirty = false;
```

开发扩展时要面向 Turbowarp 不要面向 Gandi IDE，Gandi IDE 和 Turbowarp 大部分逻辑大差不差，但是在开发体验上 Turbowarp 完爆 Gandi IDE（Turbowarp 有 sourcemap，打开 DevTools 能看到实现混淆前的源代码，而不是像你这样的 `e t` 命名）。